lib/features/auth/signup/views/agreement_screen.dart

```diff
@@ -0,0 +1,119 @@
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+import 'package:provider/provider.dart';
+import 'package:cogo/features/auth/signup/views_model/agreement_view_model.dart';
+
+class AgreementScreen extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return ChangeNotifierProvider(
+      create: (_) => AgreementViewModel(),
+      child: Scaffold(
+        backgroundColor: Colors.white,
+        body: Column(
+          mainAxisAlignment: MainAxisAlignment.end,
+          children: [
+            Container(
+              padding: EdgeInsets.all(20),
+              decoration: BoxDecoration(
+                color: Colors.white,
+                borderRadius: BorderRadius.only(
+                  topLeft: Radius.circular(20),
+                  topRight: Radius.circular(20),
+                ),
+              ),
+              child: Consumer<AgreementViewModel>(
+                builder: (context, viewModel, child) => Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    _buildCheckBoxRow(
+                      context,
+                      0,
+                      '서비스 이용약관 (필수)',
+                    ),
+                    _buildCheckBoxRow(
+                      context,
+                      1,
+                      '개인정보 수집 및 이용 동의(필수)',
+                    ),
+                    _buildCheckBoxRow(
+                      context,
+                      2,
+                      '만 14세 이상 (필수)',
+                    ),
+                    _buildCheckBoxRow(
+                      context,
+                      3,
+                      '서비스 알림 수신 (필수)',
+                    ),
+                    _buildCheckBoxRow(
+                      context,
+                      4,
+                      '서비스 혜택 정보 수신 (선택)',
+                    ),
+                    SizedBox(height: 20),
+                    SizedBox(
+                      width: double.infinity,
+                      height: 50,
+                      child: ElevatedButton(
+                        onPressed: viewModel.isAllRequiredChecked
+                            ? () {
+                          // signup_screen_phone 페이지로 이동
+                          context.go('/agreement/phone');
+                        }
+                            : null,
```

Comment:
이것 자체로 viewmodel에서 액션을 지정해주면 좋을 것 같아요
예를 들어 onTap
```suggestion
                        onPressed: viewModel.onTapSignupPhone,
```
이런 식으로 정의하고 viewmodel에서 해당 액션을 구현하면 나중에 액션이 변경되었을 때, 뷰가 아니라 viewmodel만 건드려주면 좋겠죠~?